src/rpc/mining.cpp

```diff
@@ -132,7 +132,8 @@ static UniValue generateBlocks(const CScript& coinbase_script, int nGenerate, ui
             continue;
         }
         std::shared_ptr<const CBlock> shared_pblock = std::make_shared<const CBlock>(*pblock);
-        if (!ProcessNewBlock(Params(), shared_pblock, true, nullptr))
+        CValidationState state;
+        if (!ProcessNewBlock(Params(), shared_pblock, state, true, nullptr) || !state.IsValid())
```

Comment:
I don't think `|| !state.IsValid()` is necessary. `ProcessNewBlock()` cannot return true if state is invalid.

Reply:
Might very well be #16323 bleeding in, but hopefully that is required soon :).